website/documentation/tutorials/oidc-login/_index.md

```diff
@@ -0,0 +1,209 @@
+
+    ...
+
+    contexts:
+    - context:
+        cluster: shoot--project--mycluster
+        user: my-oidc
+      name: shoot--project--mycluster
+
+    ...
+
+    users:
+    - name: my-oidc
+      user:
+        auth-provider:
+          config:
+            client-id: <Client ID>
+            client-secret: <Client Secret>
+            idp-issuer-url: "https://<Issuer>/"
+            extra-scopes: email,offline_access,profile
+          name: oidc
+    ``` 
+
+4. Ensure that the modified context is the active context `current-context: shoot--project--mycluster`.
+
+## Configure the shoot cluster
+
+Modify the shoot cluster YAML as follows:
+
+```yaml
+kind: Shoot
+apiVersion: garden.sapcloud.io/v1beta1
+metadata:
+  name: mycluster
+  namespace: garden-project
+...
+spec:
+  kubernetes:
+    kubeAPIServer:
+      oidcConfig:
+        clientID: <Client ID>
+        issuerURL: "https://<Issuer>/"
+        usernameClaim: email
+```
+
+This change of the `Shoot` manifest triggers a reconciliation. Once the reconciliation is finished, your OIDC configuration is applied. It does *not* invalidate other certificate based authentication methods. Wait for Gardener to reconcile the change. It can take up to 5 minutes.
+
+
+## Authorize an authenticated user
+
+For simplicity, we authorize a single user with the all encompassing cluster role `cluster-admin`:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: cluster-admin-test
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
+subjects:
+- apiGroup: rbac.authorization.k8s.io
+  kind: User
+  name: test@test.com
+```
+
+As administrator, apply the above cluster role binding for `test@test.com`.
+
+## Verify the Result
+
+1. Activate the prepared `kubeconfig-oidc` and perform a login:
+   
+    ```
+    $ export KUBECONFIG=~/.kube/config-oidc
+    $ kubectl oidc-login
+    Open http://localhost:8000 for authentication
+    ```
+
+    The plugin opens a browser for an interctive authentication session, and in parallel serves a local webserver for the configured callback.
+
+    ![login](./login.png)
+
+2. Enter your login credentials. If you successfully verified your user the console will display the validity of your returned token:
+  
+    ```
+    You got a valid token until 2019-08-14 06:26:49 +0200 CEST
+    ```
+
+3. Inspect the `kubeconfig-oidc`. You will find two additional parameters:
+   
+    ```yaml
+    ...
+    users:
+    - name: my-oidc
+      user:
+        auth-provider:
+          config:
+            client-id: <Client ID>
+            client-secret: <Client Secret>
+            idp-issuer-url: "https://<Issuer>/"
+            extra-scopes: email,offline_access,profile
+            id-token: eyJ0eX ... 4In0.QQKS ... TTTw
+            refresh-token: LFt ... 0Skj
+          name: oidc
+    ```
+
+    The plugin persisted the `id-token` and `refresh-token` in your configuration file. 
```

Comment:
Those steps are not needed / could be rewritten. With the newest version of [kubelogin](https://github.com/int128/kubelogin#setup) you can do `kubectl get pods` and then you are automatically authenticated in the browser if you haven't been authenticated yet.